core/src/main/java/org/geysermc/geyser/item/type/DebugStickItem.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.item.type;
+
+import org.cloudburstmc.nbt.NbtMap;
+import org.cloudburstmc.nbt.NbtMapBuilder;
+import org.cloudburstmc.nbt.NbtType;
+import org.geysermc.geyser.session.GeyserSession;
+import org.geysermc.geyser.text.ChatColor;
+import org.geysermc.geyser.text.MinecraftLocale;
+import org.geysermc.geyser.translator.item.BedrockItemBuilder;
+
+import java.util.ArrayList;
+
+public class DebugStickItem extends Item {
+    public DebugStickItem(String javaIdentifier, Builder builder) {
+        super(javaIdentifier, builder);
+    }
+
+    @Override
+    public NbtMap build(GeyserSession session, BedrockItemBuilder builder) {
+        NbtMapBuilder nbt = builder.getOrCreateNbt();
+
+        // No enchantments -> no glint! Let's add it.
+        if (!nbt.containsKey("ench")) {
+            nbt.putList("ench", NbtType.COMPOUND, new ArrayList<>());
+        }
+
+        // Only make the name pink - WHICH IS DONE CLIENT SIDE o.o - if no custom name exists
+        String translation = MinecraftLocale.getLocaleString("item.minecraft.debug_stick", session.locale());
+
+        if ((ChatColor.RESET + ChatColor.WHITE + translation).equals(builder.getCustomName())) {
```

Comment:
I get why we do this... wonder if we should prevent Geyser from originally setting this custom name in the first place?

Reply:
The issue is that the name translation happens fully independently from the java item. Additionally, the text color is not actually sent, this is done client side - hence, I'm checking here if it's the "default" result and overriding it here. It must be checked, as a custom color/text *can* be set which overrides the client default of light purple.